src/voice-search/voice-search.ts

```diff
@@ -0,0 +1,162 @@
+import {
+  Component,
+  Input,
+  Inject,
+  forwardRef,
+  NgZone,
+  ContentChild,
+  ElementRef,
+  TemplateRef,
+  OnInit,
+} from '@angular/core';
+
+import { connectVoiceSearch } from 'instantsearch.js/es/connectors';
+import { VoiceListeningState } from 'instantsearch.js/es/lib/voiceSearchHelper';
+import { BaseWidget } from '../base-widget';
+import { NgAisInstantSearch } from '../instantsearch/instantsearch';
+import { noop } from '../utils';
+
+type TemplateContext = {
+  status: string;
```

Comment:
```js
type Status =
  | 'initial'
  | 'askingPermission'
  | 'waiting'
  | 'recognizing'
  | 'finished'
  | 'error';

type TemplateContext = {
  status: Status;
  ...
```

This looks good. If we type it that way, we no longer need to declare constants like this

```js
const STATUS_INITIAL = 'initial';
const STATUS_ASKING_PERMISSION = 'askingPermission';
const STATUS_WAITING = 'waiting';
const STATUS_RECOGNIZING = 'recognizing';
const STATUS_FINISHED = 'finished';
const STATUS_ERROR = 'error';
```

right?

Instead of `if (state === STATUS_ERROR)`, we can just code `if (state === 'error')` and typo will be caught by TS.

Can you think of any drawback on this approach? I haven't typed like that so far.

Reply:
No drawbacks I'm aware of. @samouss may have an input on this?

Reply:
Nope it should be fine. You'll have intellisense for free!

Reply:
6da770b